1) A company decided to give bonus of 5% to employee if his/her year of service is more than 5 years.
Ask user for their salary and year of service and print the net bonus amount.

In [1]:
salary = float(input("Enter your Salary ??"))
services_years = int(input("enter your services years ??"))

if services_years > 5:
    bonus = salary * .05
    print(f"congratulations! you are eligilble for 5% bonus : {bonus}")
else:
    print(f"sorry you are not eligbile for bonus")

congratulations! you are eligilble for 5% bonus : 1750.0



2) Write a program to check whether a person is eligible for voting or not. (accept age from user) if age is greater than 17 eligible otherwise not eligible

In [2]:
user_age = int(input("please enter your age ??"))

if user_age > 17 :
    print(f"congratulation! you are eligible for vote.")
else:
    print(f"sorry! you are not eligible")


congratulation! you are eligible for vote.



3) Write a program to check whether a number entered by user is even or odd.

In [3]:
#Write a program to check whether a number entered by user is even or odd.

even_number = int(input("please enter the number ??"))

if even_number % 2 == 0:
    print(f"the number you enter {even_number} is even")
else:
    print(f"the number you enter {even_number} is odd")


the number you enter 20 is even


4) Write a program to check whether a number is divisible by 7 or not.
Show Answer

In [4]:
num = int(input("please enter the number ??"))

if num % 7 == 0:
    print(f"the number you enter {num} is divisible")
else:
    print(f"the number you enter {num} is not divisible")

the number you enter 35 is divisible


5) Write a program to display 
"Hello" if a number entered by user is a multiple of five , otherwise print "Bye".

In [5]:
num = int(input("please enter number ??"))

if num % 5 == 0:
    print("hello")
else:
    print("by")

hello


7) Write a program to display the last digit of a number.


In [6]:
num = int(input("please enter the number ??"))

last_digit = num % 10
print(f"the last digit of {num} is {last_digit}")

the last digit of 350 is 0


9) Take values of length and breadth of a rectangle from user and print if it is square or rectangle.

In [7]:
lenght = float(input("please enter lenght"))
breadth = float(input("please enter breadth"))

if lenght == breadth:
    print("it is a square")
else:
    print("it is a rectangle")


it is a square


10) Take two int values from user and print greatest among them.

In [8]:
value_one = int(input("please enter value"))
value_two = int(input("please enter value"))

if value_one > value_two:
    print(f"value one {value_one} is greater than value two {value_two}")
else:
    print(f"value one {value_one} is not greater than value two {value_two}")


value one 35 is greater than value two 10


11) A shop will give discount of 10% if the cost of purchased quantity is more than 1000.
Ask user for quantity
Suppose, one unit will cost 100.
Judge and print total cost for user.

In [9]:
unit_price = 100
quantity = int(input("please enter the quantity"))

total_cost = unit_price*quantity

if total_cost > 1000:
    discount = total_cost *.01
    total_cost -= discount
    print(f"you received 10% discount of {discount:.2f}!")
print(f"total cost to be paid : {total_cost:.2f}")


you received 10% discount of 35.00!
total cost to be paid : 3465.00


12) A school has following rules for grading system:

a. Below 25 - F

b. 25 to 45 - E

c. 45 to 50 - D

d. 50 to 60 - C

e. 60 to 80 - B

f. Above 80 - A

Ask user to enter marks and print the corresponding grade.

In [11]:
marks = int(input("Enter your Marks:"))

if marks > 80 and marks <= 100 :
      print("Grade A")

elif marks > 60 and marks <= 80:
      print("Grade B")

elif marks > 50 and marks <= 60:
      print("Grade c")
elif marks > 45 and marks <= 50:
      print("Grade d")
elif marks > 25 and marks <= 45:
      print("Grade e")
elif marks > 0 and marks <= 25:
      print("Grade f")
else:
      print("Invalid input")

Grade B


14)A student will not be allowed to sit in exam if his/her attendence is less than 75%.

Take following input from user

Number of classes held

Number of classes attended.

And print

percentage of class attended

Is student is allowed to sit in exam or not.


In [12]:
classes_held = int(input("Enter the total number of classes held: "))
classes_attended = int(input("Enter the number of classes attended: "))

# Calculate attendance percentage
attendance_percentage = (classes_attended / classes_held) * 100

# Display attendance percentage
print(f"Attendance Percentage: {attendance_percentage:.2f}%")

# Check if student is allowed to sit in the exam
if attendance_percentage >= 75:
    print("The student is allowed to sit in the exam.")
else:
    print("The student is NOT allowed to sit in the exam.")



Attendance Percentage: 99.00%
The student is allowed to sit in the exam.


15) Modify the above question to allow student to sit if he/she has medical cause. Ask user if he/she has medical cause or not ( 'Y' or 'N' ) and print accordingly.

In [13]:
classes_held = int(input("Enter the total number of classes held: "))
classes_attended = int(input("Enter the number of classes attended: "))

# Calculate attendance percentage
attendance_percentage = (classes_attended / classes_held) * 100

# Display attendance percentage
print(f"Attendance Percentage: {attendance_percentage:.2f}%")

# Check if attendance is less than 75%
if attendance_percentage < 75:
    # Ask if the student has a medical cause
    medical_cause = input("Do you have a medical cause? (Y/N): ").strip().upper()

    if medical_cause == 'Y':
        print("The student is allowed to sit in the exam due to medical cause.")
    else:
        print("The student is NOT allowed to sit in the exam.")
else:
    print("The student is allowed to sit in the exam.")


Attendance Percentage: 99.00%
The student is allowed to sit in the exam.


16) Write a program to check if a year is leap year or not.

If a year is divisible by 4 then it is leap year but if the year is century year like 2000, 1900, 2100 then it must be divisible by 400.

In [14]:
year = int(input("Enter a year: "))

# Check if it's a leap year
if (year % 4 == 0 and year % 100 != 0) or (year % 400 == 0):
    print(f"{year} is a Leap Year.")
else:
    print(f"{year} is NOT a Leap Year.")


2025 is NOT a Leap Year.


17) Ask user to enter age, gender ( M or F ), marital status ( Y or N ) and then using following rules print their place of service.

if employee is female, then she will work only in urban areas.

if employee is a male and age is in between 20 to 40 then he may work in anywhere

if employee is male and age is in between 40 t0 60 then he will work in urban areas only.

And any other input of age should print "ERROR"

In [15]:
age = int(input("Enter your age: "))
gender = input("Enter your gender (M/F): ").strip().upper()
marital_status = input("Are you married? (Y/N): ").strip().upper()

# Check the conditions for place of service
if gender == 'F':
    print("You will work only in urban areas.")
elif gender == 'M':
    if 20 <= age <= 40:
        print("You may work anywhere.")
    elif 40 <= age <= 60:
        print("You will work only in urban areas.")
    else:
        print("ERROR: Invalid age for employment.")
else:
    print("ERROR: Invalid gender input.")


You may work anywhere.


6) Write a program to calculate the electricity bill (accept number of unit from user) according to the following criteria :
     Unit                                                     Price  
uptp 100 units                                             no charge
Next 200 units                                              Rs 5 per unit
After 200 units                                             Rs 10 per unit
(For example if input unit is 350 than total bill amount is Rs.3500
(For example if input unit is 97 than total bill amount is Rs.0
(For example if input unit is 150 than total bill amount is Rs.750

13) Take input of age of 3 people by user and determine oldest and youngest among them.

In [16]:
units = int(input("Enter the number of units consumed: "))

# Initialize total bill amount
bill_amount = 0

# Calculate the bill based on unit consumption
if units <= 100:
    bill_amount = 0  # No charge for first 100 units
elif units <= 300:
    bill_amount = (units - 100) * 5  # Rs. 5 per unit for next 200 units
else:
    bill_amount = (200 * 5) + ((units - 300) * 10)  # Rs. 5 for 200 units, Rs. 10 beyond 300 units

# Display the total bill amount
print(f"Total bill amount is Rs. {bill_amount}")



Total bill amount is Rs. 1000


In [17]:
# Take input of age of 3 people by user and determine oldest and youngest among them.
# Accept ages from user
age1 = int(input("Enter the age of first person: "))
age2 = int(input("Enter the age of second person: "))
age3 = int(input("Enter the age of third person: "))

# Determine the oldest and youngest
oldest = max(age1, age2, age3)
youngest = min(age1, age2, age3)

# Display results
print(f"The oldest person is {oldest} years old.")
print(f"The youngest person is {youngest} years old.")


The oldest person is 55 years old.
The youngest person is 25 years old.
